### ISSUE TITLE
Shapiro-Wilk reports N = 41,188 for a statistic computed on a 5,000-row subsample

### ISSUE THREAD

**Issue by AshishK714** [social-engineer-ai/PyAnalytica#8]
**Labels:** bug, severity:medium, analyze, reporting

## Environment

| | |
|---|---|
| PyAnalytica | 0.7.0 (installed from `dist/pyanalytica-0.7.0-py3-none-any.whl` into a clean venv) |
| Python | 3.14 |
| Key deps | shiny 1.7.0, pandas 3.0.5, numpy 2.5.2, scipy 1.18.1, matplotlib 3.11.1, seaborn 0.13.2 |
| OS / browser | Windows 11, Chrome |
| Dataset | `bank_campaign.csv` — 41,188 rows x 16 cols (UCI Bank Marketing, columns renamed) |

Found during a systematic 153-case sweep of every panel against one dataset.
Numbers were checked against an independent pandas/scipy/sklearn computation, not just
"did it render".

## Summary

`shapiro_wilk_test` correctly works around scipy's N > 5000 limitation by testing a seeded
random sample of 5,000 rows — but the UI reports **N = 41,188** next to the resulting W, with
nothing indicating the test was subsampled.

## Steps to reproduce

1. Load `bank_campaign.csv`.
2. Analyze > Means > **Normality (Shapiro-Wilk)**, Numeric Variable = `age`. Run Test.

## Actual

- Interpretation: *"The distribution of age significantly deviates from normality
  (W = 0.9614, p < .001)."*
- Results table: **N = 41188**, skewness 0.7847, kurtosis 0.7911.

The statistic 0.9614 comes from a 5,000-row sample. Run on the full column, scipy gives
W = 0.9572 (and warns "For N > 5000, computed p-value may not be accurate").

## Cause

`pyanalytica/analyze/normality.py::shapiro_wilk_test` — the subsampling is deliberate and
documented in the docstring:

```python
if n > 5000:
    rng = np.random.RandomState(42)
    sample = rng.choice(data, size=5000, replace=False)
```

but `n` reported in `NormalityResult` is `len(data)`, i.e. the full column, and neither the
interpretation string nor the table mentions the sample.

## Why this matters

The statistical choice is right; the reporting is not. A student writes up "Shapiro-Wilk on
41,188 observations, W = 0.9614", which is false, and the number is not reproducible from the
stated n.

## Suggested fix

Report the tested n, and say so in the interpretation:

> W = 0.9614, p < .001 — computed on a random sample of 5,000 of 41,188 rows
> (Shapiro-Wilk is unreliable above n = 5000).

## Evidence

`shots/D12_DEFECT_shapiro_N.jpg`

---

*Found in a systematic 153-case sweep of every panel against a 41,188-row teaching dataset, with each number checked against an independent pandas/scipy/scikit-learn computation rather than against the app itself.*
